### DISCUSSION TITLE
git add issue

### DISCUSSION THREAD

**Discussion by ashokkumarsoliyappan** [gitpython-developers/GitPython#1139] [Q&A] [answered]
Hi ,
I am new to Python and Git. Found this library to run Git commands using Python. I am trying to clone an already created repository on Google Cloud to my local directory on windows. and i try to add the repository by using the command 
`repo.git.add("automationPuml")`

but i'm getting the error 
```
    raise GitCommandError(command, status, stderr_value, stdout_value)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git add automationPuml
  stderr: 'fatal: pathspec 'automationPuml' did not match any files'
```

**Comment by Byron** [comment, marked as answer]
It's difficult to provide a concise answer given I don't truly understand the question. However, under the assumption that you are not trying to add a submodule (i.e. 'repository') to the local clone using `git add` but instead try to add a file, then the error message indicates that `automationPuml` does not exist on disk and thus could not be added.

It's exactly the same result as running `git add automationPuml` on the command-line, since you are [running git directly](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly) using GitPython as proxy.

I am closing the issue as it's not representing anything that could be fixed, and further comments are welcome on the closed issue as well.
